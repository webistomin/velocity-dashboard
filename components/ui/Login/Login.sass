+b('login')
  display: flex
  flex-direction: column
  flex-grow: 1

  +e('container')
    display: inherit
    flex-direction: inherit
    flex-grow: inherit
    justify-content: center
    padding-bottom: rem(52px)

  +e('title')
    text-align: center

    +media($screen-sm)
      text-align: left

  +e('desc')
    text-align: center

    +media($screen-sm)
      text-align: left

  +e('heading')
    margin-bottom: rem(48px)

    +media($screen-sm)
      margin-bottom: rem(30px)

    +media($screen-lg)
      margin-bottom: rem(45px)

  +e('form-inputs')
    margin-bottom: rem(25px)

  +e('form-group')
    &:not(:last-of-type)
      margin-bottom: rem(20px)

  +e('submit')
    margin-bottom: rem(14px)

  +e('text')
    text-align: center

  +e('changer')
    margin-left: 0.5ch

  +e('col')
    +m('left')
      position: relative

      +media($screen-sm)
        width: calc(66vw - 14%)
        display: flex
        flex-direction: column

        +media($screen-lg)
          width: 41%

    +m('right')
      display: none

      +media($screen-sm)
        display: block
        position: fixed
        top: 0
        right: 0
        width: 32vw
        height: 100vh

      +media($screen-lg)
        width: 50%

  +e('holder')
    backface-visibility: hidden
    will-change: opacity, position

    +media($screen-sm)
      width: rem(320px)
      margin: auto

  +e('list')
    display: grid
    grid-template-columns: repeat(auto-fill, minmax(46%, 1fr))
    grid-gap: 20px
    margin-bottom: rem(20px)

  +e('option-name')
    font: var(--text-button)
    padding-bottom: rem(6px)

  +e('option-desc')
    margin-bottom: auto

  +e('option-content')
    min-height: rem(196px)
    box-sizing: border-box
    padding: rem(20px)
    display: flex
    height: 100%
    flex-direction: column
    justify-content: center
    align-items: center
    text-align: center

  +e('option-icon')
    margin-bottom: rem(15px)

  +e('inner')
    +media($screen-sm)
      display: flex
      height: 100%
      flex-grow: 1

  +e('video')
    width: 100%
    height: 100%
    display: block
    object-fit: cover

  +e('video-block')
    width: 100%
    height: 100%
    position: relative
    background-image: url('/img/login-poster.jpg')
    background-size: cover
    background-position: center
    background-repeat: no-repeat

    &::before
      content: ''
      z-index: 2
      position: absolute
      top: 0
      left: 0
      width: 100%
      height: 100%
      background-color: var(--color-primary)
      opacity: 0.2

