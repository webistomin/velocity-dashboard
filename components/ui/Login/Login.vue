<template lang="pug">
  section.login
    .login__container.container
      .login__inner
        .login__col.login__col_left
          BaseTitle(:level='3').login__title Sign in to Velocity
          p.login__desc.paragraph.paragraph_color_darkgray Please enter your credentials to proceed.
          form.login__form
            .login__form-inputs
              BaseFormGroup(
                :type='"email"'
                :label='"Email address"'
                :id='"sign-in-email"'
                :placeholder='"Enter your email"'
                :validator='$v.signInForm.email'
                @input='signInForm.email = $event'
                :value='signInForm.email'
                @blur='$v.signInForm.email.$touch()'
              ).login__form-group
              BaseFormGroup(
                :type='"password"'
                :label='"Password"'
                :id='"sign-in-password"'
                :placeholder='"Enter your password"'
                :validator='$v.signInForm.password'
                @input='signInForm.password = $event'
                :value='signInForm.password'
                @blur='$v.signInForm.password.$touch()'
              ).login__form-group
            BaseButton(
              :type='"submit"'
            ) Sign in
        .login__col.login__col_right
</template>

<script lang="ts">
  import Vue from 'vue';
  import Component from 'vue-class-component';
  // @ts-ignore
  import { email, required } from 'vuelidate/lib/validators';
  import BaseTitle from 'components/base/BaseTitle';
  import BaseFormGroup from 'components/base/BaseFormGroup';
  import BaseButton from 'components/base/BaseButton';

  @Component({
    name: 'Login',
    components: { BaseTitle, BaseFormGroup, BaseButton },
    validations: {
      signInForm: {
        email: {
          required,
          email,
        },
        password: {
          required,
        },
      },
    },
  })
  export default class Login extends Vue {
    public data() {
      return {
        signInForm: {
          email: '',
          password: '',
        },
      };
    }
  }
</script>

<style lang="sass">
  @import "Login"
</style>
