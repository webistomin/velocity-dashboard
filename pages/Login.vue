<template lang="pug">
  h1 Start
</template>

<script lang="ts">
  import Vue from 'vue';

  export default Vue.extend({
    name: 'Login',
  });
</script>
