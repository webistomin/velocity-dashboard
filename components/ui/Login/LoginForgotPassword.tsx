import { VueComponent } from 'types/vue-components';
import { Component, Emit } from 'nuxt-property-decorator';
import { VNode } from 'vue';
import { email, required } from 'vuelidate/lib/validators';
import { IUserInterface } from 'common/types/user/user-schema';
import { serverUrls } from 'common/urls/serverUrls';
import { IAuthResetResponseBody } from 'common/types/auth/reset';

import BaseTitle from 'components/base/BaseTitle';
import BaseFormGroup from 'components/base/BaseFormGroup';
import BaseButton from 'components/base/BaseButton';
import BaseLink from 'components/base/BaseLink';
import { FormTypes, ILoginProps } from './Login';

export interface ILoginForgotForm {
  email: IUserInterface['email'];
}

@Component({
  name: 'LoginForgot',
  validations: {
    forgotForm: {
      email: {
        required,
        email,
      },
    },
  },
})
export default class LoginForgot extends VueComponent<ILoginProps> {
  public forgotForm: ILoginForgotForm = {
    email: '',
  };

  public isLoading: boolean = false;

  @Emit('setFormName')
  public setFormType(type: FormTypes): FormTypes {
    return type;
  }

  public async onSubmit(event: Event) {
    event.preventDefault();

    const validator = this.$v;
    validator.$touch();

    if (!validator.$anyError) {
      this.isLoading = true;

      try {
        const data = this.forgotForm;
        await this.$axios.$post(serverUrls.auth.reset, data).then((response: IAuthResetResponseBody) => {
          this.isLoading = false;
          if (response.success) {
            this.$notify({
              group: 'auth',
              type: 'success',
              title: 'Success',
              text: 'Link was sent to your email',
              duration: 3000,
            });
          }
        });
      } catch (e) {
        this.isLoading = false;
        this.$notify({
          group: 'auth',
          type: 'error',
          title: 'Authentication error',
          text: e?.response?.data?.message,
          duration: 3000,
        });
      }
    }
  }

  public render(): VNode {
    return (
      <div class='login__holder'>
        <div class='login__heading'>
          <BaseTitle class='login__title' level={3}>
            Recover password
          </BaseTitle>
          <p class='login__desc paragraph paragraph_color_darkgray'>Don't worry, happens to the best of us.</p>
        </div>
        <form class='login__form' onSubmit={this.onSubmit}>
          <div class='login__form-inputs'>
            <BaseFormGroup
              class='login__form-group'
              type='email'
              label='Email address'
              id='forgot-email'
              placeholder='Enter your email'
              validator={this.$v.forgotForm.email}
              onInput={($event: string) => (this.forgotForm.email = $event)}
              value={this.forgotForm.email}
              onBlur={this.$v.forgotForm.email?.$touch}
            />
          </div>
          <BaseButton class='login__submit' type='submit' isLoading={this.isLoading}>
            Reset my password
          </BaseButton>
          <p class='login__text paragraph'>
            Back to
            <BaseLink class='login__changer' isButton={true} onClick={() => this.setFormType(FormTypes.SIGN_IN)}>
              Sign in
            </BaseLink>
          </p>
        </form>
      </div>
    );
  }
}
