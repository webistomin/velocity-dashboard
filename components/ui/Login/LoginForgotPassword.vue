<template lang="pug">
  .login__holder
    .login__heading
      BaseTitle(:level='3').login__title Recover password
      p.login__desc.paragraph.paragraph_color_darkgray Don't worry, happens to the best of us.
    form.login__form
      .login__form-inputs
        BaseFormGroup(
          :type='"email"'
          :label='"Email address"'
          :id='"forgot-email"'
          :placeholder='"Enter your email"'
          :validator='$v.forgotForm.email'
          @input='forgotForm.email = $event'
          :value='forgotForm.email'
          @blur='$v.forgotForm.email.$touch()'
        ).login__form-group
      BaseButton(
        :type='"submit"'
      ).login__submit Reset my password
      p.login__text.paragraph Back to
        BaseLink(
          :isButton='true'
          @click='setFormType(FormTypes.SIGN_IN)'
          ).login__changer Sign in
</template>

<script lang="ts">
  import Vue from 'vue';
  import Component from 'vue-class-component';
  // @ts-ignore
  import { email, required } from 'vuelidate/lib/validators';
  import BaseFormGroup from 'components/base/BaseFormGroup';
  import BaseButton from 'components/base/BaseButton';
  import BaseLink from 'components/base/BaseLink';
  import { FormTypes } from 'components/ui/Login/Login.vue';

  @Component({
    name: 'LoginForgot',
    components: { BaseFormGroup, BaseButton, BaseLink },
    validations: {
      forgotForm: {
        email: {
          required,
          email,
        },
      },
    },
  })
  export default class LoginForgot extends Vue {
    FormTypes: any = FormTypes;

    public data() {
      return {
        forgotForm: {
          email: '',
        },
      };
    }

    public setFormType(type: FormTypes): void {
      this.$emit('setFormName', type);
    }
  }
</script>
