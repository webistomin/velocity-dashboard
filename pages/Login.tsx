import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import Login from 'components/ui/Login';

@Component({
  name: 'LoginPage',
  layout: 'login',
})
export default class LoginPage extends VueComponent {
  render(): VNode {
    return (
      <main class='page-content'>
        <Login />
      </main>
    );
  }
}
