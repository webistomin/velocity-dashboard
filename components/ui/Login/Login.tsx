import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import { VueComponent } from 'types/vue-components';

import LoginSignIn from './LoginSignIn';
import LoginSignUp from './LoginSignUp';
import LoginForgotPassword from './LoginForgotPassword';
import LoginVideo from './LoginVideo';
import LoginHeader from './LoginHeader';

import './Login.sass';

export enum FormTypes {
  SIGN_IN = 'sign-in',
  SIGN_UP = 'sign-up',
  FORGOT_PASSWORD = 'forgot-password',
}

export interface ILoginProps {
  onSetFormName: (type: FormTypes) => void;
}

@Component({
  name: 'Login',
})
export default class Login extends VueComponent {
  public currentForm = FormTypes.SIGN_IN;

  public setFormType(formName: FormTypes): void {
    this.currentForm = formName;
  }

  public render(): VNode {
    return (
      <section class='login'>
        <h2 class='visually-hidden'>Authorization</h2>
        <div class='login__container container'>
          <div class='login__inner'>
            <div class='login__col login__col_left'>
              <LoginHeader />
              <transition name='login-fade' mode='out-in'>
                {this.currentForm === FormTypes.SIGN_IN && (
                  <LoginSignIn onSetFormName={(formName) => this.setFormType(formName)} />
                )}
                {this.currentForm === FormTypes.SIGN_UP && (
                  <LoginSignUp onSetFormName={(formName) => this.setFormType(formName)} />
                )}
                {this.currentForm === FormTypes.FORGOT_PASSWORD && (
                  <LoginForgotPassword onSetFormName={(formName) => this.setFormType(formName)} />
                )}
              </transition>
            </div>
            <div class='login__col login__col_right'>
              <LoginVideo />
            </div>
          </div>
        </div>
      </section>
    );
  }
}
