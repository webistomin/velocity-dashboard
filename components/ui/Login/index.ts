import Login from './Login.vue';
import LoginSignIn from './LoginSignIn.vue';
import LoginSignUp from './LoginSignUp.vue';
import LoginForgotPassword from './LoginForgotPassword.vue';

export { Login, LoginSignIn, LoginSignUp, LoginForgotPassword };

export default Login;
