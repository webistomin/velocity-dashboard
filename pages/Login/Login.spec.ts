import { shallowMount } from '@vue/test-utils';
import Login from './Login';

describe('Login page', () => {
  const factory = (propsData = {}, options = {}) => {
    return shallowMount(Login, {
      propsData: {
        ...propsData,
      },
      stubs: ['notifications'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
