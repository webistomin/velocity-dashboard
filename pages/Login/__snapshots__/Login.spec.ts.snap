// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Matches snapshot 1`] = `
"<main class=\\"page-content\\">
  <login-stub></login-stub>
  <notifications-stub group=\\"auth\\" position=\\"top left\\" animation-name=\\"v-popup-fade-left\\"></notifications-stub>
</main>"
`;
